src/btree/bt_page.c

```diff
@@ -21,6 +21,89 @@ static int __inmem_row_leaf_entries(WT_SESSION_IMPL *, const WT_PAGE_HEADER *, u
 WT_STAT_USECS_HIST_INCR_FUNC(internal_reconstruct, perf_hist_internal_reconstruct_latency)
 WT_STAT_USECS_HIST_INCR_FUNC(leaf_reconstruct, perf_hist_leaf_reconstruct_latency)
 
+/*
+ * __page_find_min_delta --
+ *     Find the smallest key among all current delta entries. When keys are equal across deltas,
+ *     prefer the latest delta (higher index). Also advance older duplicates so only the latest
+ *     delta for a given key is emitted.
+ */
+static inline int
+__page_find_min_delta(WT_SESSION_IMPL *session, WT_CELL_UNPACK_DELTA_INT **unpacked_deltas,
+  size_t *delta_size_each, size_t *delta_idx, size_t delta_count,
+  WT_CELL_UNPACK_DELTA_INT **min_delta, uint32_t *min_d)
+{
+    WT_DECL_RET;
+    WT_ITEM dk, mk;
+    size_t d;
+    int cmp;
+
+    *min_delta = NULL;
+    *min_d = UINT32_MAX;
+
+    /* First pass: pick the smallest key among available deltas. Prefer later delta if equal. */
+    for (d = 0; d < delta_count; ++d) {
+        /* Skip exhausted delta arrays */
+        if (delta_idx[d] >= delta_size_each[d])
+            continue;
+
+        WT_ASSERT(session, unpacked_deltas[d] != NULL);
+
+        dk.data = unpacked_deltas[d][delta_idx[d]].key.data;
+        dk.size = unpacked_deltas[d][delta_idx[d]].key.size;
+
+        if (*min_delta == NULL) {
+            /* First available delta entry */
+            *min_delta = &unpacked_deltas[d][delta_idx[d]];
+            *min_d = (uint32_t)d;
+            continue;
+        }
+
+        mk.data = (*min_delta)->key.data;
+        mk.size = (*min_delta)->key.size;
+
+        WT_ERR(__wt_compare(session, S2BT(session)->collator, &dk, &mk, &cmp));
+
+        if (cmp < 0) {
+            /* Found a strictly smaller key */
+            *min_delta = &unpacked_deltas[d][delta_idx[d]];
+            *min_d = (uint32_t)d;
+        } else if (cmp == 0) {
+            /* Same key: prefer latest delta (higher index) */
+            if ((uint32_t)d > *min_d) {
+                *min_delta = &unpacked_deltas[d][delta_idx[d]];
+                *min_d = (uint32_t)d;
+            }
+        }
+    }
+
+    /* Second pass: advance other deltas that have the same key as the chosen min_delta */
+    if (*min_delta != NULL) {
```

Comment:
I think we can do this in a single pass if we iterate from latest to earliest.

Reply:
Fixex in c5e9462cc9aaf108a28e6df48bf35724b4aa93e3 to iterate from the latest to the earliest.